### PR TITLE
🏗️ Change admin UI files to React structure

### PR DESCRIPTION
**Pull request by shellyYG**

### CODE REVIEW

File: likecoin/js/admin-settings/src/components/DropDown.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        <label>{props.title}</label>
+      </th>
+      <td>
+        <select
+          ref={props.selectRef}
+          onChange={handleSelect}
+          value={props.selected}
+        >
+          {props.options.map((option) => (
+            <option value={option.value}> {option.label} </option>
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/components/DropDown.js:17]: Should add key for `<option/>`

File: likecoin/js/admin-settings/src/components/MainSettingTable.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    DBSiteLikerIdEnabled: select(
+      SITE_LIKER_INFO_STORE_NAME,
+    ).selectSiteLikerInfo().DBSiteLikerIdEnabled,
+    DBDisplayOptionSelected: select(
+      SITE_LIKER_INFO_STORE_NAME,
+    ).selectSiteLikerInfo().DBDisplayOptionSelected,
+    DBPerPostOptionEnabled: select(
+      SITE_LIKER_INFO_STORE_NAME,
+    ).selectSiteLikerInfo().DBPerPostOptionEnabled,
+  }));
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/components/MainSettingTable.js:50]: Is it equivalent to this? 
```suggestion
  } = useSelect((select) => select(SITE_LIKER_INFO_STORE_NAME).selectSiteLikerInfo());
```

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+import { USER_LIKER_INFO_STORE_NAME } from '../store/user-likerInfo-store';
+
+function LikecoinButtonPage() {
+  const {
+    DBSiteLikerId,
+    DBSiteLikerAvatar,
+    DBSiteLikerDisplayName,
+    DBSiteLikerWallet,
+    DBSiteLikerIdEnabled,
+  } = useSelect((select) => ({
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:24]: Same here

File: likecoin/js/admin-settings/src/pages/PublishSettingPage.js [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+    DBSiteMattersAutoPublish: select(
+      SITE_MATTERS_STORE_NAME,
+    ).selectSiteMattersOptions().DBSiteMattersAutoPublish,
+    DBSiteMattersAddFooterLink: select(
+      SITE_MATTERS_STORE_NAME,
+    ).selectSiteMattersOptions().DBSiteMattersAddFooterLink,
+    DBISCNBadgeStyleOption: select(
+      SITE_MATTERS_STORE_NAME,
+    ).selectSiteMattersOptions().DBISCNBadgeStyleOption,
+  }));
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/PublishSettingPage.js:46]: Same here
```suggestion
  } = useSelect((select) => select(SITE_MATTERS_STORE_NAME).selectSiteMattersOptions());
```

File: likecoin/js/admin-settings/src/components/MainSettingTable.js (hunk 2 above)
Comment (shellyYG) [reply likecoin/js/admin-settings/src/components/MainSettingTable.js:50]: true! much clear, thanks

File: likecoin/js/admin-settings/src/pages/PublishSettingPage.js [hunk 5]
```diff
@@ -19,7 +19,6 @@ function PublishSettingPage() {
   // eslint-disable-next-line arrow-body-style
   const {
     DBSiteMattersId,
-    DBSiteMattersToken,
```
Comment (williamchong) [inline likecoin/js/admin-settings/src/pages/PublishSettingPage.js]: Why is this removed?

File: likecoin/js/admin-settings/src/store/web-monetization-store.js [hunk 6]
```diff
@@ -37,7 +37,7 @@ const actions = {
 };
 
 const selectors = {
-  selectPaymentPointer: (state) => state.pointer || '',
+  selectPaymentPointer: (state) => state,
```
Comment (williamchong) [inline likecoin/js/admin-settings/src/store/web-monetization-store.js:40]: What does this change do?

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 7]
```diff
@@ -68,6 +70,7 @@ function LikecoinButtonPage() {
     () => debounce(async (likerId) => {
       setSavedSuccessful(false);
       setIsLoading(true);
+      if (!likerId) return; // user did not input any value. Want to forceDisconnect.
```
Comment (williamchong) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:73]: Why not just set `forceDisconnect` to true/false here?

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 8]
```diff
@@ -112,24 +117,23 @@ function LikecoinButtonPage() {
...
+      data.userLikerInfos.avatar = '';
+    } else {
+      data.userLikerInfos.likecoin_id = likerDisplayName === '-' ? DBUserLikerId : likerIdValue;
+      data.userLikerInfos.display_name = likerDisplayName === '-' ? DBUserLikerDisplayName : likerDisplayName;
+      data.userLikerInfos.wallet = likerDisplayName === '-' ? DBUserLikerWallet : likerWalletAddress;
+      data.userLikerInfos.avatar = likerDisplayName === '-' ? DBUserLikerAvatar : likerAvatar;
+    }
     try {
+      // Change notice
+      if (!data.userLikerInfos.likecoin_id) setForceDisconnect(false); // has insert liker ID
```
Comment (williamchong) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:136]: Why `likercoin_id` not exists means `has insert liker ID`?

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 9]
```diff
@@ -112,24 +117,23 @@ function LikecoinButtonPage() {
...
+    if (forceDisconnect) {
+      data.userLikerInfos.likecoin_id = '';
+      data.userLikerInfos.display_name = '';
+      data.userLikerInfos.wallet = '';
+      data.userLikerInfos.avatar = '';
+    } else {
+      data.userLikerInfos.likecoin_id = likerDisplayName === '-' ? DBUserLikerId : likerIdValue;
+      data.userLikerInfos.display_name = likerDisplayName === '-' ? DBUserLikerDisplayName : likerDisplayName;
+      data.userLikerInfos.wallet = likerDisplayName === '-' ? DBUserLikerWallet : likerWalletAddress;
+      data.userLikerInfos.avatar = likerDisplayName === '-' ? DBUserLikerAvatar : likerAvatar;
```
Comment (williamchong) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:132]: I think a much better way is to rewrite this to split the display data (`-` for empty values) and actual data (`''` or `null` for empty values) in local state of this component.
In such design you no need to handle `-` as a data value or add `forceDisconnect` as a hacky flag, just handle empty data as disconnect would do

**Review by williamchong — CHANGES_REQUESTED** [review]

File: likecoin/js/admin-settings/src/store/web-monetization-store.js (hunk 6 above)
Comment (shellyYG) [reply likecoin/js/admin-settings/src/store/web-monetization-store.js:40]: The state is now called DBPaymentPointer, so state.pointer will become undefined.

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js (hunk 7 above)
Comment (shellyYG) [reply likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:73]: It was to enable the hacky way for empty likerId value. Will change this.

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js (hunk 8 above)
Comment (shellyYG) [reply likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js]: Changed it in the simplify UI logic commit.

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js (hunk 9 above)
Comment (shellyYG) [reply likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js]: Changed it in the simplify UI logic commit.

File: likecoin/admin/metabox.php [hunk 10]
```diff
@@ -186,15 +186,219 @@ function likecoin_get_meta_box_publish_params( $post, $force = false ) {
...
+								<?php esc_html_e( 'Submit to ISCN', LC_PLUGIN_SLUG ); ?>
+							</button>
+						</span>
+						<?php } else { ?>
+							<span id="lcISCNPublish" style="display:<?php echo esc_attr( empty( $status['ipfs']['url'] ) ? 'none' : '' ); ?>">
+								<button id="lcISCNPublishBtn" class="button button-primary">
+									<?php esc_html_e( 'Submit to ISCN', LC_PLUGIN_SLUG ); ?>
+								</button>
+							</span>
+					<?php } ?>
```
Comment (nwingt) [inline likecoin/admin/metabox.php:282]:
```suggestion
					<?php } else { ?>
						<span id="lcISCNPublish" style="display:<?php echo esc_attr( empty( $status['ipfs']['url'] ) ? 'none' : '' ); ?>">
							<button id="lcISCNPublishBtn" class="button button-primary">
								<?php esc_html_e( 'Submit to ISCN', LC_PLUGIN_SLUG ); ?>
							</button>
						</span>
					<?php } ?>
```

File: likecoin/js/admin-settings/src/components/LikecoinInfoTable.js [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+                    </td>
+                    <td>
+                      {!props.isLoading && (
+                        <Text text={props.likerDisplayName ? props.likerDisplayName : '-'} />
+                      )}
+                    </td>
+                    <td>
+                      {!props.isLoading && (
+                        <Text text={props.likerWalletAddress ? props.likerWalletAddress : '-'} />
+                      )}
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/components/LikecoinInfoTable.js:82]:
```suggestion
                      {!props.isLoading && (
                        <Text text={props.likerDisplayName || '-'} />
                      )}
                    </td>
                    <td>
                      {!props.isLoading && (
                        <Text text={props.likerWalletAddress || '-'} />
                      )}
```

File: likecoin/js/admin-settings/src/components/MainSettingTable.js [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+            handleLikerIdInputChange={handleLikerIdInputChange}
+            handleDisconnect={handleDisconnect}
+            editable={true}
+            isMainSettingPage={true}
+            showChangeButton={true}
+            showDisconnectButton={false}
+          />
+        ) : (
+          ''
+        )}
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/components/MainSettingTable.js:199]:
```suggestion
        {siteLikerIdEnabled && (
          <LikecoinInfoTable
            likerIdValue={likerIdValue}
            likerDisplayName={likerDisplayName}
            likerWalletAddress={likerWalletAddress}
            likerAvatar={likerAvatar}
            isLoading={isLoading}
            isChangingTypingLiker={isChangingTypingLiker}
            handleClickOnChange={handleClickOnChange}
            handleLikerIdInputChange={handleLikerIdInputChange}
            handleDisconnect={handleDisconnect}
            editable={true}
            isMainSettingPage={true}
            showChangeButton={true}
            showDisconnectButton={false}
          />
        )}
```

File: likecoin/js/admin-settings/src/components/MattersDescription.js [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+            src="https://matters.news/static/icon-144x144.png"
+            alt="matters-logo"
+          ></img>
+        </a>
+        {__('What is Matters.news?', 'likecoin')}
+      </h2>
+      <p></p>
+      <p>{localizedMattersShortIntro}</p>
+      <p>{localizedMattersLongIntro}</p>
+      <br />
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/components/MattersDescription.js:51]:
```suggestion
```

File: likecoin/js/admin-settings/src/components/MattersDescription.js [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+    {
+      IPFS: createElement(Link, {
+        text: __('IPFS', 'likecoin'),
+        linkAddress: 'https://ipfs.io',
+      }),
+    },
+  );
+  return (
+    <div style={{ textAlign: 'left' }}>
+      <p></p>
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/components/MattersDescription.js:32]:
```suggestion
```

File: likecoin/js/admin-settings/src/components/MattersDescription.js [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+          <img
+            height="32"
+            weight="32"
+            src="https://matters.news/static/icon-144x144.png"
+            alt="matters-logo"
+          ></img>
+        </a>
+        {__('What is Matters.news?', 'likecoin')}
+      </h2>
+      <p></p>
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/components/MattersDescription.js:48]:
```suggestion
```

File: likecoin/js/admin-settings/src/components/ParagraphTitle.js [hunk 16]
```diff
@@ -0,0 +1,10 @@
+function ParagraphTitle(props) {
+  return (
+    <p>
+      <strong>{props.text}</strong>
+      <br />
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/components/ParagraphTitle.js:5]:
```suggestion
```

File: likecoin/js/admin-settings/src/components/WebMonetizationDescription.js [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+      linkAddress: 'https://community.webmonetization.org/likecoinprotocol',
+    }),
+  },
+);
+function WebMonetizationDescription() {
+  return (
+    <div style={{ textAlign: 'left' }}>
+      <p></p>
+      <h2>{__('What is Web Monetization?', 'likecoin')}</h2>
+      <p></p>
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/components/WebMonetizationDescription.js:38]:
```suggestion
      <h2 style="margin: 20px 0;">{__('What is Web Monetization?', 'likecoin')}</h2>
```

File: likecoin/js/admin-settings/src/components/Text.js [hunk 18]
```diff
@@ -0,0 +1,5 @@
+function Text(props) {
+  return <p ref={props.testRef}> {props.text}</p>;
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/components/Text.js:2]:
```suggestion
  return <p ref={props.testRef}>{props.text}</p>;
```

File: likecoin/js/admin-settings/src/components/SubmitButton.js [hunk 19]
```diff
@@ -0,0 +1,13 @@
+import { __ } from '@wordpress/i18n';
+
+function SubmitButton() {
+  return (
+    <tr style={{ textAlign: 'left' }}>
+      <button className="likecoinSubmitButton">
+        {__('Confirm', 'likecoin')}
+      </button>
+    </tr>
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/components/SubmitButton.js:9]: Why wrapping `<button/>` with `<tr/>`?

File: likecoin/js/admin-settings/src/components/Section.js [hunk 20]
```diff
@@ -0,0 +1,9 @@
+function Section(props) {
+  return (
+    <tr>
+      <h2 style={{ textAlign: 'left' }}>{props.title}</h2>
+    </tr>
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/components/Section.js:5]: Why wrapping `<h2/>` with `<tr/>`

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 21]
```diff
@@ -0,0 +1,214 @@
...
+  const defaultLikerDisplayName = siteLikerIdEnabled
+    ? DBSiteLikerDisplayName
+    : DBUserLikerDisplayName;
+  const defaultLikerWalletAddress = siteLikerIdEnabled
+    ? DBSiteLikerWallet
+    : DBUserLikerWallet;
+  const defaultLikerAvatar = siteLikerIdEnabled
+    ? DBSiteLikerAvatar
+    : DBUserLikerAvatar;
+  const [likerIdValue, getLikerIdValue] = useState(defaultLikerId);
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:50]: Naming `get` is confusing for the function that update the state
```suggestion
  const [likerIdValue, setLikerIdValue] = useState(defaultLikerId);
```

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 22]
```diff
@@ -0,0 +1,214 @@
...
+  const [likerDisplayName, getLikerDisplayName] = useState(
+    defaultLikerDisplayName,
+  );
+  const [likerWalletAddress, getLikerWalletAddress] = useState(
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:54]:
```suggestion
  const [likerWalletAddress, setLikerWalletAddress] = useState(
```

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 23]
```diff
@@ -0,0 +1,214 @@
...
+    defaultLikerWalletAddress,
+  );
+  const [likerAvatar, getLikerAvatar] = useState(defaultLikerAvatar);
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:57]:
```suggestion
  const [likerAvatar, setLikerAvatar] = useState(defaultLikerAvatar);
```

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 24]
```diff
@@ -0,0 +1,214 @@
...
+  const [savedSuccessful, setSavedSuccessful] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [, setIsDisconnect] = useState(false);
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:60]: What this line for if you are not using the value 🤨

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 25]
```diff
@@ -0,0 +1,214 @@
...
+    setShowChangeButton(!!defaultLikerId);
+    setShowDisconnectButton(!!defaultLikerId);
+    setHasValidLikecoinId(!!defaultLikerId);
+  }, [
+    DBSiteLikerIdEnabled,
+    defaultLikerId,
+    defaultLikerDisplayName,
+    defaultLikerWalletAddress,
+    defaultLikerAvatar,
+  ]);
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:110]: `useState` already comes with a default value that you set, what are these lines for?

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 26]
```diff
@@ -0,0 +1,214 @@
...
+    getLikerIdValue('');
+    getLikerDisplayName('');
+    getLikerWalletAddress('');
+    getLikerAvatar('');
+    setIsDisconnect(true);
+  }
+  return (
+    <div className="wrap likecoin">
+      <LikecoinHeading />
+      {!savedSuccessful && ''}
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:166]:
```suggestion
```

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 27]
```diff
@@ -0,0 +1,214 @@
...
+        />
+        <Section title={__('Your Likecoin button', 'likecoin')} />
+        {hasValidLikecoinId ? (
+          <LikeButtonPreview
+            userLikerId={likerIdValue}
+            hasValidLikecoinId={hasValidLikecoinId}
+          />
+        ) : (
+          ''
+        )}
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:207]:
```suggestion
        {hasValidLikecoinId && (
          <LikeButtonPreview
            userLikerId={likerIdValue}
            hasValidLikecoinId={hasValidLikecoinId}
          />
        )}
```

File: likecoin/js/admin-settings/src/pages/PublishSettingPage.js [hunk 28]
```diff
@@ -0,0 +1,285 @@
...
+
+  useEffect(() => {
+    setSiteMattersAutoSaveDraft(DBSiteMattersAutoSaveDraft);
+    setSiteMattersAutoPublish(DBSiteMattersAutoPublish);
+    setSiteMattersAddFooterLink(DBSiteMattersAddFooterLink);
+  }, [
+    DBSiteMattersAutoSaveDraft,
+    DBSiteMattersAutoPublish,
+    DBSiteMattersAddFooterLink,
+  ]);
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/PublishSettingPage.js:217]: Can be merged into one?

File: likecoin/js/admin-settings/src/pages/PublishSettingPage.js [hunk 29]
```diff
@@ -0,0 +1,285 @@
...
+  return (
+    <div className="wrap likecoin">
+      <LikecoinHeading />
+      {!savedSuccessful && ''}
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/PublishSettingPage.js:221]:
```suggestion
```

File: likecoin/js/admin-settings/src/pages/WebMonetizationPage.js [hunk 30]
```diff
@@ -0,0 +1,95 @@
...
+    setSavedSuccessful(false);
+  }
+  useEffect(() => {
+    setPaymentPointer(DBPaymentPointer);
+  }, [DBPaymentPointer]);
+
+  return (
+    <div className="wrap likecoin">
+      <LikecoinHeading />
+      {!savedSuccessful && ''}
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/WebMonetizationPage.js:51]:
```suggestion
```

File: likecoin/js/admin-settings/src/pages/WebMonetizationPage.js [hunk 31]
```diff
@@ -0,0 +1,95 @@
...
+                  onChange={handlePaymentPointerChange}
+                ></input>{' '}
+                <a
+                  rel="noopener noreferrer"
+                  target="_blank"
+                  href="https://webmonetization.org/docs/ilp-wallets/"
+                >
+                  {__('What is payment pointer?', 'likecoin')}
+                </a>
+              </td>
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/WebMonetizationPage.js:85]: It is strange using `th` along with `td`

https://www.w3schools.com/html/html_table_headers.asp

File: likecoin/js/admin-settings/src/pages/PublishSettingPage.js (hunk 28 above)
Comment (shellyYG) [reply likecoin/js/admin-settings/src/pages/PublishSettingPage.js:217]: Yes, makes sense! Totally overlooked. :P

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js (hunk 24 above)
Comment (shellyYG) [reply likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:60]: Was a legacy code 😂

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js (hunk 25 above)
Comment (shellyYG) [reply likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:110]: To change dynamically when default values changes. But how default value changes is confusing. I will re-write that part.

File: likecoin/js/admin-settings/src/pages/WebMonetizationPage.js (hunk 31 above)
Comment (shellyYG) [reply likecoin/js/admin-settings/src/pages/WebMonetizationPage.js:85]: Agree, this is a rare usage of vertical table though: https://www.w3schools.com/html/tryit.asp?filename=tryhtml_table_th_vertical 😂

File: likecoin/js/admin-settings/src/components/SubmitButton.js (hunk 19 above)
Comment (shellyYG) [reply likecoin/js/admin-settings/src/components/SubmitButton.js:9]: so button can be in the left while inner text is align-center

Comment (shellyYG) [reply likecoin/js/admin-settings/src/components/SubmitButton.js:9]: changed it from <tr /> to <div />

File: likecoin/js/admin-settings/src/components/Section.js (hunk 20 above)
Comment (shellyYG) [reply likecoin/js/admin-settings/src/components/Section.js:5]: Yes, this one is redundant, can delete

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 32]
```diff
@@ -0,0 +1,214 @@
...
+      setSavedSuccessful(false);
+      if (!likerId) return;
+      try {
+        const response = await axios.get(
+          `https://api.like.co/users/id/${likerId}/min`,
+        );
+        getLikerIdValue(response.data.user);
+        getLikerDisplayName(response.data.displayName);
+        getLikerWalletAddress(response.data.cosmosWallet);
+        getLikerAvatar(response.data.avatar);
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js]: have changed prefix to `set`

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 33]
```diff
@@ -0,0 +1,214 @@
...
+import Section from '../components/Section';
+import LikecoinInfoTable from '../components/LikecoinInfoTable';
+import SubmitButton from '../components/SubmitButton';
+import LikeButtonPreview from '../components/LikeButtonPreview';
+import SettingNotice from '../components/SettingNotice';
+import LikecoinHeading from '../components/LikecoinHeading';
+import { SITE_LIKER_INFO_STORE_NAME } from '../store/site-likerInfo-store';
+import { USER_LIKER_INFO_STORE_NAME } from '../store/user-likerInfo-store';
+
+function LikecoinButtonPage() {
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:17]: After revisiting this, I think it is better to separate this component into a pure component and inject the data from the store through props using a higher-order component

**Review by nwingt — CHANGES_REQUESTED** [review]

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js (hunk 33 above)
Comment (nwingt) [reply likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:17]: Those useEffects are confusing to me

Comment (nwingt) [reply likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:17]: By injecting through props, those `useEffects` can be omitted

Comment (shellyYG) [reply likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:17]: As discussed. Changed in the next comment. :)

**Review by williamchong — APPROVED** [review]

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js [hunk 34]
```diff
@@ -0,0 +1,234 @@
...
+  return (
+    <PureLikecoinButtonPage
+      DBSiteLikerIdEnabled={DBSiteLikerIdEnabled}
+      defaultLikerId={defaultLikerId}
+      defaultLikerDisplayName={defaultLikerDisplayName}
+      defaultLikerWalletAddress={defaultLikerWalletAddress}
+      defaultLikerAvatar={defaultLikerAvatar}
+      postUserLikerInfo={postUserLikerInfo}
+    />
+  );
```
Comment (nwingt) [inline likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js:231]: No need to use `useState` here as you will not change the state later

```
<PureLikecoinButtonPage
  defaultLikerId={siteLikerIdEnabled ? DBSiteLikerId : DBUserLikerId}
  ...
/>
```

You may also use `useMemo` to cache the value to improve performance 
```
const defaultLikerId = useMemo(
  () => siteLikerIdEnabled ? DBSiteLikerId : DBUserLikerId,
  [
    siteLikerIdEnabled,
    DBSiteLikerId,
    DBUserLikerId,
  ]
)
```

**Review by nwingt — CHANGES_REQUESTED** [review]

File: likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js (hunk 34 above)
Comment (shellyYG) [reply likecoin/js/admin-settings/src/pages/LikecoinButtonPage.js]: Cool that it can be simplified like this. Changed below.

**Review by nwingt — APPROVED** [review]
